python/foxglove-sdk/python/foxglove/channel.py

```diff
@@ -32,7 +32,7 @@ def __init__(
             encoding is presumed to be "json".
         :param schema: A definition of your schema. Pass a :py:class:`Schema`
             for full control. If a dictionary is passed, it will be treated as a
-            JSON schema.
+            JSON schema. If not provided, the message_encoding must be "json".
```

Comment:
This is a bit backwards in my view - the message encoding (or more specifically the channel setup) is what decides whether you need a schema to deserialize not the other way around.